rocketpy/Function.py

```diff
@@ -2024,9 +2024,267 @@ def integral(self, a, b, numerical=False):
+            A Function object which gives the derivative of self.
+        """
+        # Check if Function object source is array
+        if isinstance(self.source, np.ndarray):
+            # Operate on grid values
+            Ys = np.diff(self.source[:, 1]) / np.diff(self.source[:, 0])
+            Xs = self.source[:-1, 0] + np.diff(self.source[:, 0]) / 2
+            source = np.concatenate(([Xs], [Ys])).transpose()
+
+            # Retrieve inputs, outputs and interpolation
+            inputs = self.__inputs__[:]
+            outputs = "d(" + self.__outputs__[0] + ")/d(" + inputs[0] + ")"
+            outputs = "(" + outputs + ")"
+            interpolation = "linear"
+
+            # Create new Function object
+            return Function(source, inputs, outputs, interpolation)
+        else:
+            return Function(lambda x: self.differentiate(x))
+
+    def integralFunction(self, lower=None, upper=None, datapoints=100):
+        """Returns a Function object representing the integral of the Function object.
+
+        Parameters
+        ----------
+        lower : scalar, optional
+            The lower limit of the interval in which the function is to be
+            plotted. If the Function is given by a dataset, the default
+            value is the start of the dataset.
+        upper : scalar, optional
+            The upper limit of the interval in which the function is to be
+            plotted. If the Function is given by a dataset, the default
+            value is the end of the dataset.
+        datapoints : int, optional
+            The number of points in which the integral will be evaluated for
+            plotting it, which draws lines between each evaluated point.
+            The default value is 100.
+
+        Returns
+        -------
+        result : Function
+            The integral of the Function object.
+        """
+        # Check if lower and upper are given
+        if lower is None:
+            if isinstance(self.source, np.ndarray):
+                lower = self.source[0, 0]
+            else:
+                raise ValueError("Lower limit must be given if source is a function.")
+        if upper is None:
+            if isinstance(self.source, np.ndarray):
+                upper = self.source[-1, 0]
+            else:
+                raise ValueError("Upper limit must be given if source is a function.")
+
+        # Create a new Function object
+        xData = np.linspace(lower, upper, datapoints)
+        yData = np.zeros(datapoints)
+        for i in range(datapoints):
+            yData[i] = self.integral(lower, xData[i])
+        return Function(np.concatenate(([xData], [yData])).transpose(), 
+                    inputs=self.__inputs__, 
+                    outputs=[o + " Integral" for o in self.__outputs__])
+
+    def invert(self, lower=None, upper=None, datapoints=100):
+        """
+        Inverts the Function. Makes the domain the range and the range the domain.
```

Comment:
To follow the pattern stablished by `derivativeFunction` and `integralFunction`, consider renaming this method to `inverseFunction`. This, together with a rewording of the docstring makes it clear that the method generates a new Function, instead of performing the operation in place.

```suggestion
    def inverseFunction(self, lower=None, upper=None, datapoints=100):
        """
        Returns the inverse of the Function. The inverse function of F is a function that undoes the operation of F. The 
        inverse of F exists if and only if F is bijective. Makes the domain the range and the range the domain.
```